ui/StatusQ/src/StatusQ/Components/StatusTimeStampLabel.qml

```diff
@@ -12,7 +12,24 @@ StatusBaseText {
     color: Theme.palette.baseColor1
     font.pixelSize: 10
     visible: !!text
-    text: showFullTimestamp ? LocaleUtils.formatDateTime(timestamp) : LocaleUtils.formatRelativeTimestamp(timestamp)
+    text: d.formattedLabel
+
+    QtObject {
+        id: d
+        property string formattedLabel: showFullTimestamp ? LocaleUtils.formatDateTime(root.timestamp) : LocaleUtils.formatRelativeTimestamp(root.timestamp)
+
+        readonly property var _conn: Connections {
+            target: SharedUpdateTimer
+            enabled: root.visible && root.timestamp && !root.showFullTimestamp
+            function onTriggered() {
+                const newFormattedLabel = LocaleUtils.formatRelativeTimestamp(root.timestamp)
+                if (newFormattedLabel !== d.formattedLabel) {
+                    d.formattedLabel = newFormattedLabel
+                }
+            }
+        }
+    }
```

Comment:
assuming that `SharedUpdateTimer` exposes additional read-only helper property `secondsElapsed`
the code can be simplified to sth like that (`formattedLabelChanged` won't be emitted unnecessarily as well):

```suggestion
    QtObject {
        id: d
        
        property string formattedLabel: LocaleUtils.formatDateTime(root.timestamp) 
        
        Binding on formattedLabel {
            when: !showFullTimestamp
            
            value: {
                SharedUpdateTimer.secondsElapsed
                return LocaleUtils.formatRelativeTimestamp(root.timestamp)
            }
        }
    }
```